### ISSUE TITLE
Selected nodes do not move when dragged after selecting with `Shift+Click` or programaticaly.

### ISSUE THREAD

**Issue by etrobert** [xyflow/xyflow#4441]
### What platform were you using when you found the bug?

- React Flow / Svelte Flow version: 12.0.1
- Browser and version: Arc, Chrome
- OS and version: MacOS 14.5

### Live code example

https://codesandbox.io/p/sandbox/react-flow-starter-ts-r69228

### Describe the Bug

Selecting multiple nodes by using `Shift+Click` or by programmatically changing the `selected` state does not create a "selection box" and most importantly does not trigger `onSelectionDrag` when trying to move the nodes (only the node under the cursor moves).

When holding `Shift` and dragging to select we get the expected behavior, we visually see a selection box and `onSelectionDrag` is called when dragging any of the nodes.

### Steps to reproduce the bug or issue

1. Go to any flow with multiple nodes
2. Left click on a node to select it
3. Use `Shift+Click` on another node to select it
4. Drag one of the nodes

### Expected behavior

As a user, I expect all of the selected nodes to move but instead only the node under the cursor moves.

As a user, I expect `onSelectionDrag` to be called but instead only `onNodeDrag` for the node under the cursor is called.

As a user, I expect a selection box to appear around all of the nodes that I selected, but no selection box appears.

### Screenshots or Videos

https://github.com/user-attachments/assets/b22307a3-f938-4da7-84ef-417433961b2e

### Additional context

In our codebase, we have bound a `Ctrl-A` keyboard shortcut to select all nodes, which sets their `selected` attribute to `true`. This correctly sets their individual selected styling but does not display the selection box and does not allow selection drag.

**Comment by moklick** [comment]
I agree that this would be the better UX. We will discuss it with the team and see what we can do about this.

**Comment by MatheusRoichman** [comment]
@moklick do you have any updates? :)
